llvm/lib/Transforms/Scalar/DeadStoreElimination.cpp

```diff
@@ -1916,6 +1916,7 @@ struct DSEState {
     Updater.insertDef(NewAccessMD, /*RenameUses=*/true);
     Updater.removeMemoryAccess(Malloc);
     Malloc->replaceAllUsesWith(Calloc);
+    EI.removeInstruction(Malloc);
```

Comment:
Should this use deleteDeadInstruction() instead? This can probably run into the same BatchAA invalidation issue, right?

Reply:
yeah that's better, updated, thanks!